lib/json-schema/json-schema-validator.ts

```diff
@@ -0,0 +1,99 @@
+var jsv = require("jsv").JSV;
+
+export class JsonSchemaValidator implements IJsonSchemaValidator {
+	private static ENVIRONMENT_ID = "json-schema-draft-03";
+	private static DEFAULT_SCHEMA_URI = "http://json-schema.org/draft-03/schema#";
+
+	private environment: any = null;
+	private _availableProperties: any = null;
+	private _validProperties: IStringDictionary = null;
+
+	constructor(private $errors: IErrors,
+		private $frameworkProjectResolver: Project.IFrameworkProjectResolver,
+		private $jsonSchemaLoader: IJsonSchemaLoader, // Don't delete this row, we need it
+		private $jsonSchemaResolver: IJsonSchemaResolver) {
+		this.environment = jsv.createEnvironment(JsonSchemaValidator.ENVIRONMENT_ID);
+		this.environment.setDefaultSchemaURI(JsonSchemaValidator.DEFAULT_SCHEMA_URI);
+	}
+
+	public getValidProperties(framework: string, frameworkVersion: string): IStringDictionary {
+		if(!this._validProperties) {
+			this._validProperties = Object.create(null);
+
+			_.each(this.getAvailableProperties(framework, frameworkVersion), (propertyName: string) => {
+				this._validProperties[propertyName.toLowerCase()] = propertyName;
+			});
+		}
+
+		return this._validProperties;
+	}
+
+	public validate(data: IProjectData): any {
+		var validationErrors = this.getValidationErrors(data);
+		if(_.keys(validationErrors).length !== 0) {
+			var output = _.values(validationErrors).join("\n");
+			this.$errors.fail(output);
+		}
+	}
+
+	public isValid(data: IProjectData): boolean {
+		var errors = this.getValidationErrors(data);
+		return _.keys(errors).length !== 0;
+	}
+
+	private getValidationErrors(data: IProjectData): IStringDictionary {
+		var validationSchema = this.tryResolveValidationSchema(data.FrameworkVersion, data.Framework);
+		var schema = this.environment.createSchema(validationSchema);
+		var validationResult = this.environment.validate(data, schema);
+		var errors = validationResult.errors;
+		var result: IStringDictionary = Object.create(null);
+
+		_.each(errors, (error: any) => {
+			var schemaUri = error.schemaUri;
+			var schemaUriParts = schemaUri.split("/");
+			var propertyName = schemaUriParts[schemaUriParts.length - 1];
+			result[propertyName] = util.format("Property %s: %s %s", propertyName, error.message, error.details);
+		});
+
+		return result;
+	}
+
+	private getAvailableProperties(framework: string, frameworkVersion: string): any {
+		if(!this._availableProperties) {
+			var schema = this.tryResolveValidationSchema(framework, frameworkVersion);
+			this._availableProperties = _.union(_.keys(schema.properties), _.keys(schema.extends.properties));
+		}
+		return this._availableProperties;
+	}
+
+	private tryResolveValidationSchema(frameworkVersion: string, framework: string): ISchema {
+		var validationSchemaName = this.getValidationSchemaName(frameworkVersion, framework);
+		var schema = this.$jsonSchemaResolver.getSchema(validationSchemaName);
+
+		if(!schema) {
+			this.$errors.fail("Unable to resolve validation schema.");
```

Comment:
Is there anything the user can do to fix the issue? If so, we can add it to the message.